### PR TITLE
fix: MoonPay enhancements/fixes

### LINKED ISSUE #2400: 'amount' field doesn't support comma's when buying via moonpay
### Bug description
The IOS app I use displays a comma in the on-screen keyboard when entering the amount of IOTA you want to order via Moonpay. In my region it is common to use a comma as decimal separator (netherlands). The mask in the amount field can't cope with this input, since it gives NaN for several fields that use this number as input. See screenshot.

Trinity issue

### Bug description
A general description of the bug.

### Environment Specs
Apple Iphone 8
OS version 13.2
Trinity version: 1.2.0 (95)

### Steps To Reproduce
1. Open trinity, it opens in the balance screen
2. tap "buy IOTA"
3. Confirm the trinity account you want to use
This opens up the screen where you can input the amount of currency you would like to purchase. My keyboard shows a comma (see screenshot). When entering a normal number, the values are updated in realtime. When entering the comma, the fields go to NaN and $0

### Expected behaviour
The software should be capable of processing the given value with the buttons that are supplied on the on-screen keyboard.
1. Display only a dot or a comma, if this is possible
2. Convert a comma to a dot in the background
3. Make the calculation so that it accepts both inputs as correct

### Actual behaviour
NaN shows when a comma is entered

### Errors
Screenshot provided

### LINKED ISSUE #2477: Select Country listed instead of Select State in MoonPay
### Bug description

Select Country is listed twice of in MoonPay on desktop.

### LINKED ISSUE #2478: MoonPay add payment card layout issue
### Bug description

- Remember credit card settings checkbox not visible on some resolutions

### Environment Specs

- OS version: Windows 10
- Dell XPS laptop connected to a 27" screen with 3200 x 1800 recommended resolution with size of text, app, other set 150% (default).

- Trinity version: 1.2.0

unknown-1

### LINKED ISSUE #2480: Incorrect date of birth formatting in MoonPay
### Bug description

Entering a DOB with incorrect formatting provides no error prompt

### Environment Specs
What device are you running Trinity on?

- OS version: Windows 10

What build of Trinity are you running? 

- Trinity version: 1.2.0

### LINKED ISSUE #2481: MoonPay incorrectly displays USA
### Bug description

USA is not yet supported for IOTA purchases by MoonPay but it shows in the wallet.

### Environment Specs

Trinity version: 1.2.0

### PR DESCRIPTION
**Pull request by rajivshah3**
# Description

Fixes and enhancements to MoonPay registration and purchase process:
- Use placeholder for DOB field
- Validate DOBs
- Convert commas to periods when used as decimal separators
- Hide US from country list until US support is added
- Fix crash on country dropdown list
- Minor fixes to translation strings

Fixes #2400 
Fixes #2480 
Fixes #2477 
Fixes #2478 
Fixes #2481 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

- Tested on iOS simulator

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes
- [ ] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

File: src/mobile/src/ui/views/wallet/exchanges/MoonPay/UserBasicInfo.js [hunk 1]
```diff
@@ -150,10 +150,16 @@ class UserBasicInfo extends React.Component {
             );
         }
 
+        const dobMoment = moment(this.state.dateOfBirth, 'DD/MM/YYYY');
+
+        if (!dobMoment.isValid()) {
+            return this.props.generateAlert('error', 'Invalid DOB', 'Invalid DOB');
```
Comment (rajivshah3) [inline src/mobile/src/ui/views/wallet/exchanges/MoonPay/UserBasicInfo.js]: To do: Add translation string

File: src/mobile/src/ui/views/wallet/exchanges/MoonPay/AddAmount.js [hunk 2]
```diff
@@ -562,7 +578,7 @@ class AddAmount extends Component {
                                 </Text>
                                 <Text style={[styles.infoTextLight, textColor]}>
                                     {this.getStringifiedFiatAmount(
-                                        getAmountInFiat(Number(amount), denomination, exchangeRates),
+                                        getAmountInFiat(Number(this.parseAmount(amount)), denomination, exchangeRates),
                                     )}
                                 </Text>
                             </View>
```
Comment (rajivshah3) [inline src/mobile/src/ui/views/wallet/exchanges/MoonPay/AddAmount.js]: To do: Fee will still sometimes show as `0` when using commas

File: src/mobile/src/ui/views/wallet/exchanges/MoonPay/AddAmount.js [hunk 3]
```diff
@@ -416,7 +432,7 @@ class AddAmount extends Component {
         } = this.props;
```
Comment (rajivshah3) [inline src/mobile/src/ui/views/wallet/exchanges/MoonPay/AddAmount.js]: To do: Amount should be validated with something like `Number(this.parseAmount(amount)) !== NaN`

**Comment by cvarley100** [comment]
Is there a reason this is mobile only?

**Comment by rajivshah3** [comment]
I happened to start on mobile and was planning to do desktop afterwards

File: src/desktop/src/ui/views/exchanges/MoonPay/ReviewPurchase.js [hunk 4]
```diff
@@ -79,10 +81,24 @@ class ReviewPurchase extends React.PureComponent {
...
+            hasAnyPaymentCards,
+        } = this.props;
 
         return (
-            <form onSubmit={() => arePurchasesSuspended ? history.push('/exchanges/moonpay/purchase-suspended-warning'): createTransaction}>
+            <form
+                onSubmit={() =>
+                    arePurchasesSuspended
+                        ? history.push('/exchanges/moonpay/purchase-suspended-warning')
+                        : createTransaction
```
Comment (laumair) [inline src/desktop/src/ui/views/exchanges/MoonPay/ReviewPurchase.js]: Why aren't we calling `createTransaction` here? If purchases aren't suspended, form submission won't do anything.

**Review by laumair — CHANGES_REQUESTED** [review]

**Review by laumair — APPROVED** [review]
Looks good to me! Nice work.

**Pull request merged**